Refactoring User Login to use JWT Tokens

Astral8 opened the pull request:
Fixing methods/tests for User object so that it can utilize JWT Tokens.

Astral8 committed:
Refactored method/test for JWT Validity

Astral8 committed:
cleaned imports

astherath commented on tests/test_attempt_user_login.py:
```diff
@@ -74,7 +75,9 @@ def check_user_login_response_valid(response: HTTPResponse) -> bool:
 def check_jwt_valid(response: HTTPResponse) -> bool:
     response_dict = response.json()
-    return "jwt" in response_dict
+    enc_string = response_dict.get('jwt')
+    valid_check = Token.check_if_valid(enc_string)
+    return valid_check
```
calling `response_dict.get()` but not handling the default `None` case means that it won't fail on this statement, but instead will fail in the `Token.check_if_valid()` call, which is unintended. 

also, if the `response.json()` _doesn't_ have a `jwt` field in it, then this will fail, instead of returning `False` as it should.

astherath commented on tests/test_attempt_user_login.py:
```diff
@@ -74,7 +75,9 @@ def check_user_login_response_valid(response: HTTPResponse) -> bool:
 # fixme: check jwt validity once we get the code to do this
```
dead fixme.

astherath commented on util/users.py:
```diff
@@ -97,6 +98,8 @@ async def check_user_password_matches(login_form: user_models.UserLoginForm,
 
 
 # fixme: change this to return a Token once we have made the class
```
dead fixme

astherath commented on util/users.py:
```diff
@@ -97,6 +98,8 @@ async def check_user_password_matches(login_form: user_models.UserLoginForm,
+async def get_auth_token_from_user_data(user: user_models.User) -> str:
+    user_id = user.get_id()
+    payload_dict = {'user_id': user_id}
+    login_response = Token.get_enc_token_str_from_dict(payload_dict)
     return login_response
```
calling this `login_response` no longer makes sense (before it was being cast into a `UserLoginResponse` object).

Something more obvious like `encoded_jwt_str` or similar would be better.

astherath reviewed (changes requested):
Solid work, has some misschecks and also should implement the token handling in `UserRegistration` endpoints/test for it.

Astral8 committed:
fixed dead comments and checks cases for jwt validity

Astral8 committed:
Changed Register User method to use JWT and added new User Authentication Object

astherath committed:
Fixed return type for auth-enabled endpoints and changed tests to pass

The `register_user` function is left untouched, instead we just wrap the
response type at the `routes/` function.

Tests have been fixed to check for valid tokens on both `login` and
`register` endpoints.

astherath committed:
Merge branch 'master' of https://github.com/SparkDevTeams/underline-backend into impl-auth-login-token

astherath committed:
implemented fixes to admin models to match users
